core/src/main/scala/kafka/zk/AdminZkClient.scala

```diff
@@ -221,15 +201,15 @@ class AdminZkClient(zkClient: KafkaZkClient) extends Logging {
       AdminUtils.assignReplicasToBrokers(allBrokers, partitionsToAdd, existingAssignmentPartition0.size,
         startIndex, existingAssignment.size)
     }
+
     val proposedAssignment = existingAssignment ++ proposedAssignmentForNewPartitions
     if (!validateOnly) {
       info(s"Creating $partitionsToAdd partitions for '$topic' with the following replica assignment: " +
         s"$proposedAssignmentForNewPartitions.")
-      // add the combined new list
-      createOrUpdateTopicPartitionAssignmentPathInZK(topic, proposedAssignment, update = true)
+
+      writeTopicPartitionAssignment(topic, proposedAssignment, isUpdate = true)
```

Comment:
I guess we validated that the following checks were redundant?

```scala
     if (partitionReplicaAssignment.values.map(_.size).toSet.size != 1)
       throw new InvalidReplicaAssignmentException("All partitions should have the same number of replicas")
 
     partitionReplicaAssignment.values.foreach(reps =>
       if (reps.size != reps.toSet.size)
        throw new InvalidReplicaAssignmentException("Duplicate replica assignment found: " + partitionReplicaAssignment)
    )
```

Reply:
Also, `Topic.validate(topic)`.

Reply:
@ijuma Yes, these checks are redundant given the checks in `validateReplicaAssignment`, which checks for duplicate replicas as well as verifying the replication factor.

I considered `Topic.validate` as well, but my reasoning was that `addPartitions` depends on topic existence, so the check is not needed. Does that seem reasonable?

Reply:
Yes, I have verified above validations while adding partitions.  
and we are checking topic existence before calling addPartitions.

Reply:
Thanks for the explanation. My concern is that `adminZkClient.addPartitions` is called directly from `TopicCommand` and other methods in `AdminZkClient` generally do their own validation. Also, I'd not be surprised if other people call some of these methods directly. At least ensuring that the topic name is valid (which is cheap), might still be worth it.